src/main/java/gyro/google/compute/AbstractSslCertificateResource.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2020, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.compute;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import com.google.api.services.compute.model.SslCertificate;
+import gyro.core.GyroException;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.validation.Regex;
+import gyro.core.validation.Required;
+import gyro.google.Copyable;
+
+public abstract class AbstractSslCertificateResource extends ComputeResource implements Copyable<SslCertificate> {
+
+    private String certificatePath;
+    private String description;
+    private String name;
+    private String privateKeyPath;
+
+    // Read-only
+    private String selfLink;
+
+    /**
+     * The full path to the file containing the certificate(s) (in PEM format). The certificate chain must be no greater than 5 certs long. The chain must include at least one intermediate cert. Only used when creating the SSL certificate. (Required)
```

Comment:
Redundant as update is not supported.
```suggestion
     * The full path to the file containing the certificate(s) (in PEM format). The certificate chain must be no greater than 5 certs long. The chain must include at least one intermediate cert. (Required)
```